lib/rubygems/gemcutter_utilities.rb

```diff
@@ -250,11 +251,28 @@ def request_with_otp(method, uri, &block)
     end
   end
 
-  def ask_otp
-    say "You have enabled multi-factor authentication. Please enter OTP code."
+  def ask_otp(credentials)
+    webauthn_url = webauthn_verification_url(credentials)
+    unless webauthn_url
+      say "You have enabled multi-factor authentication. Please enter OTP code."
+    else
+      say "You have enabled multi-factor authentication. Please enter OTP code from your security device by visiting #{webauthn_url} or your authenticator app."
```

Comment:
I'm not sure why we mention the authenticator app here, I was under the impression that the flow will be entirely browser based.

Reply:
`Authenticator app` was mentioned since the user can also input a TOTP code from their authenticator device if they like.

Perhaps we can rephrase it as so:
> You have enabled multi-factor authentication. Please enter OTP code. To authenticate using a security device, please visit #{webauthn_url}.

wdyt?